src/lib/frontend/d_cnf.ml

```diff
@@ -607,12 +607,12 @@ let mk_ty_decl (ty_c: DE.ty_cst) =
     in
     let lbs = List.rev rev_lbs in
     let record_constr = Uid.of_dolmen cstr in
-    let ty = Ty.trecord ~record_constr tyvl (Uid.of_dolmen ty_c) lbs in
+    let ty = Ty.trecord ~record_constr tyvl uid lbs in
     Cache.store_ty ty_c ty
 
   | Some (Adt { cases; _ } as adt) ->
-    Nest.add_nest [adt];
-    let uid = Uid.of_dolmen ty_c in
+    let uid = Uid.of_ty_cst ty_c in
+    Nest.attach_orders [adt];
```

Comment:
Note: I think that keeping the `let uid = Uid.of_ty_cst ty_c in` after the call to `Nest.attach_orders` allows to have an assertion in `Uid.of_ty_cst` that the type has had order tags attached.